tests/test_georaster.py

```diff
@@ -1913,3 +1926,279 @@ def test_power(self, power: float | int) -> None:
+    #              'cumsum', 'cumprod', 'quantile']
+
+    # Sorting and counting ounting with single array input:
+    # sortfuncs = ['sort', 'count_nonzero', 'unique]
+
+    # --  Third, we define the test data --
+
+    # We create two random array of varying dtype
+    width = height = 5
+    min_val = np.iinfo("int32").min
+    max_val = np.iinfo("int32").max
+    transform = rio.transform.from_bounds(0, 0, 1, 1, width, height)
+    np.random.seed(42)
+    arr1 = np.random.randint(min_val, max_val, (height, width), dtype="int32") + np.random.normal(size=(height, width))
+    arr2 = np.random.randint(min_val, max_val, (height, width), dtype="int32") + np.random.normal(size=(height, width))
+
+    # Create two random masks
+    mask1 = np.random.randint(0, 2, size=(width, height), dtype=bool)
+    mask2 = np.random.randint(0, 2, size=(width, height), dtype=bool)
+
+    # Assert that there is at least one unmasked value
+    assert np.count_nonzero(~mask1) > 0
+    assert np.count_nonzero(~mask2) > 0
+
+    @pytest.mark.parametrize("ufunc_str", ufuncs_str_1nin_1nout + ufuncs_str_1nin_2nout)  # type: ignore
+    @pytest.mark.parametrize(
+        "dtype", ["uint8", "int8", "uint16", "int16", "uint32", "int32", "float32", "float64", "longdouble"]
+    )  # type: ignore
+    @pytest.mark.parametrize("nodata_init", [None, "type_default"])  # type: ignore
+    def test_array_ufunc_1nin_1nout(self, ufunc_str: str, nodata_init: None | str, dtype: str) -> None:
+        """Test that ufuncs with one input and one output consistently return the same result as for masked arrays."""
+
+        # We set the default nodata
+        if nodata_init == "type_default":
+            nodata: int | None = _default_ndv(dtype)
+        else:
+            nodata = None
+
+        # Create Raster
+        ma1 = np.ma.masked_array(data=self.arr1.astype(dtype), mask=self.mask1)
+        rst = gr.Raster.from_array(ma1, transform=self.transform, crs=None, nodata=nodata)
+
+        # Get ufunc
+        ufunc = getattr(np, ufunc_str)
+
+        # Find the common dtype between the Raster and the most constrained input type (first character is the input)
+        com_dtype = np.find_common_type([dtype] + [ufunc.types[0][0]], [])
+
+        # Catch warnings
+        with warnings.catch_warnings():
+
+            warnings.filterwarnings("ignore", category=RuntimeWarning)
+            warnings.filterwarnings("ignore", category=UserWarning)
+
+            # Check if our input dtype is possible on this ufunc, if yes check that outputs are identical
+            if com_dtype in (str(np.dtype(t[0])) for t in ufunc.types):
+
+                # For a single output
+                if ufunc.nout == 1:
+                    assert np.ma.allequal(ufunc(rst.data), ufunc(rst).data)
+
+                # For two outputs
+                elif ufunc.nout == 2:
+                    assert np.ma.allequal(ufunc(rst.data)[0], ufunc(rst)[0].data) and np.ma.allequal(
```

Comment:
Note that you are running each function twice. You could run it twice and save the outputs instead.

Reply:
Fixed at both occurences!